### PR TITLE
Fix rd.xml for Serialization Reflection Only Mode.

### LINKED ISSUE #19067: To Fix rd.xml for Serialization Reflection Only Mode
#19006 added rd.xml to `System.Runtime.Serialization.*.ReflectionOnly.Tests`, which made many serialization tests to pass in reflection only mode on UWP. The rd.xml, however, include metadata for the entire System.Private.DataContractSerialization, which is not what we want. We need to fix the rd.xml for System.Private.DataContractSerialization to include just the necessary metadata info.

### PR DESCRIPTION
**Pull request by shmao**
Reflection based serialization needs to reflect some types or methods defined in `System.Private.DataContractSerialization.dll`. #19006 added rd.xml to System.Runtime.Serialization.*.ReflectionOnly.Tests. Many tests passed after that fix. But that fix caused the test projects to not represent actual customer scenarios.

The PR changes `System.Private.DataContractSerialization.rd.xml` to keep necessary metadata and removes the rd.xml for the test projects.

Fix #19067

### CODE REVIEW

**Comment by vitek-karas** [comment]
I'm wondering about the size implications. Can we measure the size-on-disk difference before and after this change for some app which uses the serialization?

**Comment by shmao** [comment]
@vitek-karas Sure, I can measure the size. can you please point me to the files that I should measure?

**Comment by shmao** [comment]
@vitek-karas I measured the size of the serialization test project, which heavily uses serialization , before/after the fix. The size increased 24 KB. I think we're good.

File: src/System.Private.DataContractSerialization/src/Resources/System.Private.DataContractSerialization.rd.xml [hunk 1]
```diff
@@ -10,6 +10,45 @@
           </Method>
         </Type>
         <Type Name="KeyValuePairAdapter`2" Dynamic="Required All" />
+        <!-- Reflection-based serialization requires the entries below. -->
+        <Type Name="CollectionDataContract">
+          <Type Name="CollectionDataContractCriticalHelper">
+            <Method Name="BuildCreateGenericDictionaryEnumerator{K, V}" Dynamic="Required">
+              <GenericParameter Name="K" Dynamic="Required All" />
```
Comment (morganbr) [inline src/System.Private.DataContractSerialization/src/Resources/System.Private.DataContractSerialization.rd.xml]: Are these generic parameters and the others in the file expected to be user types or internal types? If they're internal types, this change is fine. If they're user types, I'd both be concerned this won't work and suspect the directives may need to be fine-tuned for size impact.

Comment (shmao) [inline src/System.Private.DataContractSerialization/src/Resources/System.Private.DataContractSerialization.rd.xml]: They are actually not needed. I've removed them. 

I was hoping they can include the needed metadata for IEnumerable types, but it didn't work.

**Pull request merged**